util/file_reader_writer.cc

```diff
@@ -114,12 +119,17 @@ Status RandomAccessFileReader::Read(uint64_t offset, size_t n, Slice* result,
     } else {
       size_t pos = 0;
       const char* res_scratch = nullptr;
+      StopWatchNano wait_for_rate_limiter(env_);
       while (pos < n) {
         size_t allowed;
         if (for_compaction_ && rate_limiter_ != nullptr) {
+          if (env_) {
```

Comment:
we have a function `RateLimiter::IsRateLimited` that tells us which ops are rate limited. You can pass it `RateLimiter::OpType::kRead` and only use timer if it returns true. Then the common case (rate limiter only applies to writes) won't need two timers per file read.